bridges/snowbridge/pallets/system-frontend/src/lib.rs

```diff
@@ -233,12 +304,48 @@ pub mod pallet {
 			T::XcmSender::deliver(ticket)
 		}
 
+		/// Swaps a specified tip asset to Ether and then burns the resulting ether for
+		/// teleportation. Returns the amount of Ether gained if successful, or a DispatchError if
+		/// any step fails.
+		fn swap_and_burn(
+			who: AccountIdOf<T>,
+			tip_asset_location: Location,
+			ether_location: Location,
+			tip_amount: u128,
+		) -> Result<u128, DispatchError>
+		where
+			<T as frame_system::Config>::AccountId: Into<Location>,
+		{
+			tracing::error!(target: LOG_TARGET, "in swap and burn");
+			// Swap tip asset to ether
+			let swap_path = vec![tip_asset_location.clone(), ether_location.clone()];
+			let who_location: Location = who.clone().into();
+
+			let ether_gained = T::Swap::swap_exact_tokens_for_tokens(
+				who.clone(),
+				swap_path,
+				tip_amount,
+				None,
+				who.clone(),
+				true,
+			)
+			.map_err(|_| Error::<T>::SwapError)?;
```

Comment:
I would log the error here most of all actually

Reply:
Added in https://github.com/paritytech/polkadot-sdk/pull/8271/commits/5bb4ffffe36527c6e2addb31fed55101e81609fc.

Reply:
But you've still got `.map_err(|_| Error::<T>::SwapError)?;`, which means you're hiding the original error returned by `swap_exact_tokens_for_tokens`